src/unitxt/api.py

```diff
@@ -170,9 +174,17 @@ def infer(
     return_data: bool = False,
     return_log_probs: bool = False,
     return_meta_data: bool = False,
+    previous_messages: Optional[list[dict[str, str]]] = None,
```

Comment:
Hi. I tend to think that adding previous messages is rather specific (and rare) behavior which is beyond the scope of a general infer() method, instead I would inline this code where it is need.

Otherwise, we need to document it, check for cases where the previous messages or current "source"f files are a string and not list, etc...

Reply:
I think that, as unitxt is tending to primarily use OpenAI chat format, having previous messages won't be that rare of a use case. Anytime the user needs to simulate a conversation with the assistant, like in a chat or CoT, this parameter will be required. 
The verification checks to be done if `previous_messages` is not `None` are:
- `previous_messages` should be a list of messages.
-  the source of the dataset's instances should be a list of messages too. 
- we can go ahead and check that `previous messages` are valid openai chat format too, i.e. have a `source` and a `content` attributes of type string. If format is not `formats.chat_api` we should check that the source of the dataset's instances are valid message lists too.

If you still think this is not the appropriate place to set the previous messages, could you suggest another place?

Reply:
Hi. I see your points.  I'll be glad to hear @elronbandel .  Maybe given the shift to chat, it is motivated.

Reply:
Can somehow this be achieved by a formatter? Like `ChatAPIFormatWithPreviousMessages`.